app/src/script.js

```diff
@@ -1,18 +1,188 @@
-import Aragon from '@aragon/client'
+import 'core-js/stable'
+import 'regenerator-runtime/runtime'
+import { map, publishReplay } from 'rxjs/operators'
+import { of } from 'rxjs'
+import AragonApi from '@aragon/api'
 
-const app = new Aragon()
+import { addressesEqual } from './lib/web3-utils'
+import tokenAbi from './abi/token.json'
 
-const initialState = {
-  dummyValue: 0
+const INITIALIZATION_TRIGGER = Symbol('INITIALIZATION_TRIGGER')
+const ACCOUNTS_TRIGGER = Symbol('ACCOUNTS_TRIGGER')
+
+const api = new AragonApi()
+
+api.indentify('Lock')
+
+api
+  .call('token')
+  .subscribe(initialize, err =>
+    console.error(`Could not start background script execution due to the contract not loading token: ${err}`)
+  )
+
+async function initialize(tokenAddress) {
+  const tokenContract = api.external(tokenAddress, tokenAbi)
+
+  return createStore({
+    token: {
+      address: tokenAddress,
+      contract: tokenContract,
+    },
+  })
+}
+
+async function createStore(settings) {
+  // Hot observable which emits an web3.js event-like object with an account string of the current active account.
+  const accounts$ = api.accounts().pipe(
+    map(accounts => {
+      return {
+        event: ACCOUNTS_TRIGGER,
+        account: accounts[0],
+      }
+    }),
+    publishReplay(1)
+  )
+
+  accounts$.connect()
+
+  const currentBlock = await getBlockNumber()
+
+  return api.store(
+    async (state, event) => {
+      const { event: eventName, blockNumber } = event
+
+      console.log('event', event)
+      //dont want to listen for past events for now
+      //(our app state can be obtained from smart contract vars)
+      if (blockNumber && blockNumber <= currentBlock) return state
+
+      let nextState = {
+        ...state,
+      }
+
+      if (eventName === INITIALIZATION_TRIGGER) {
+        nextState = await initializeState(nextState, settings)
+      } else if (eventName === ACCOUNTS_TRIGGER) {
+        nextState = await updateConnectedAccount(nextState, event)
+      } else {
+        switch (eventName) {
+          case 'ChangeLockDuration':
+            nextState = await updateLockDuration(nextState, event)
+            break
+          case 'ChangeLockAmount':
+            nextState = await updateLockAmount(nextState, event)
+            break
+          case 'NewLock':
+            nextState = await newLock(nextState, event)
+          case 'Withdrawal':
+            nextState = await newWithdrawal(nextState, event)
+            break
+          default:
+            break
```

Comment:
The new API also has SYNC events, probably best to check out their apps but you can see them in mine here: https://github.com/empowerthedao/compound-aragon-app/blob/master/compound-aragon-app/app/src/script.js#L72

Reply:
Do you know if there's a doc with the new api functionalities?